crates/wasi-component-loader/src/extension/runtime.rs

```diff
@@ -96,56 +112,35 @@ impl ExtensionRuntime for ExtensionsWasiRuntime {
     {
         let inputs = InputList::from_iter(inputs);
         async move {
-            let Some(inner) = self.0.as_ref() else {
-                return Err(PartialGraphqlError::internal_extension_error());
-            };
-
-            let id = ExtensionPoolId::Resolver(extension_id);
+            let mut instance = self.get(ExtensionPoolId::Resolver(extension_id)).await?;
 
-            let Some(pool) = inner.instance_pools.get(&id) else {
-                return Err(PartialGraphqlError::internal_extension_error());
-            };
-
-            let mut instance = pool.get().await;
-
-            let arguments = minicbor_serde::to_vec(arguments).unwrap();
             let directive = wit::FieldDefinitionDirective {
                 name,
                 site: wit::FieldDefinitionDirectiveSite {
                     parent_type_name: field.parent_entity().name(),
                     field_name: field.name(),
-                    arguments: &arguments,
+                    arguments: minicbor_serde::to_vec(arguments).unwrap(),
```

Comment:
You should not use `minicbor_serde::to_vec` anymore. You did the fix in the crate for null values, so we now have our own `to_vec` that uses the new setting:

https://github.com/grafbase/grafbase/blob/main/crates/wasi-component-loader/src/cbor.rs

Reply:
Good point, it should never happen in practice since I'm not using serde_json::Value but we can't know that here.